src/titiler/extensions/titiler/extensions/render.py

```diff
@@ -0,0 +1,182 @@
+from titiler.core.models.OGC import Link
+
+
+class RenderItem(BaseModel, extra="allow"):
+    """Render item for stac render extension."""
+
+    assets: List[str]
+    title: Optional[str] = None
+    rescale: Optional[List[Annotated[List[float], 2]]] = None
+    nodata: Optional[float] = None
+    colormap_name: Optional[str] = None
+    colormap: Optional[Dict] = None
+    color_formula: Optional[str] = None
+    resampling: Optional[str] = None
+    expression: Optional[str] = None
+    minmax_zoom: Optional[Annotated[List[int], 2]] = None
+
+
+class RenderItemWithLinks(BaseModel):
+    """Same as RenderItem with url and params."""
+
+    params: RenderItem
+    links: List[Link]
+
+
+class RenderItemList(BaseModel):
+    """List of Render Items with links."""
+
+    renders: Dict[str, RenderItemWithLinks]
+    links: List[Link]
+
+
+@define
+class stacRenderExtension(FactoryExtension):
+    """Add /renders endpoint to a STAC TilerFactory."""
+
+    def register(self, factory: MultiBaseTilerFactory):
+        """Register endpoint to the tiler factory."""
+
+        def _prepare_query_string(render: Dict, src_path: str) -> str:
+            """Prepare render related query params.
+
+            Validates and filters query params that titiler can understand.
+            If titiler does not support parameter, it will be ignored.
+            """
+            # List of dependencies a `/tile` URL should validate
+            # Note: Those dependencies should only require Query() inputs
+            tile_dependencies = [
+                factory.layer_dependency,
+                factory.dataset_dependency,
+                # Image rendering Dependencies
+                factory.rescale_dependency,
+                factory.color_formula_dependency,
+                factory.colormap_dependency,
+                factory.render_dependency,
+            ]
+
+            query = {"url": src_path}
+            for dependency in tile_dependencies:
+                dep = get_dependant(path="", call=dependency)
+                if dep.query_params:
+                    # call the dependency with the query-parameters values
+                    query_values, _errors = request_params_to_args(
+                        dep.query_params, render
+                    )
+                    _ = dependency(**query_values)
+                    query.update(query_values)
+            return urlencode(
+                {key: value for key, value in query.items() if value is not None},
+                doseq=True,
+            )
+
+        def _prepare_render_item(
+            render_id: str, render: Dict, request: Request, src_path: str
+        ) -> Dict:
+            """Prepare single render item."""
+            query_string = _prepare_query_string(render, src_path)
+            links = [
+                {
+                    "href": factory.url_for(
+                        request,
+                        "Show STAC render",
```

Comment:
🤯 I didn't realized we could search by name

Reply:
If I am not mistaken, `url_for` always searches by name. An important caveat is that if you do not specify the name explicitly in the `@router` decorator, it would be taken from the name of a decorated function.